### PR TITLE
Make okta compatible

### PR DESCRIPTION
**Pull request by mrmicjam**
- ask for entity_id from service provicer metadata
- add extra fields required by okta for user creation

@trustsitka/dev

### CODE REVIEW

File: saml-idp.py [hunk 1]
```diff
@@ -149,11 +149,35 @@ def make_assertion(resp, acs, irt, iss, username, validity):
                attrib={'InResponseTo': irt,
                        'NotOnOrAfter': tsnow(validity),
                        'Recipient': acs})
+    # new stuff
```
Comment (danitornow) [inline saml-idp.py]: Not a blocker, maybe call this "Okta new stuff" just for later support! :)

File: saml-idp.py [hunk 2]
```diff
@@ -228,7 +252,8 @@ def root():
     relay_state = request.values.get('RelayState')
     return f'''<HTML><BODY><CENTER><H1>Login - Step 1</H1>
 <FORM ACTION="https://{HOST}:{PORT}/login" METHOD=POST>
-Username: <INPUT TYPE=TEXT VALUE="admin" NAME="username"><BR>
+Service Provider Metadata Entity ID +/-: <INPUT TYPE=TEXT NAME="sp_entity_id" VALUE="https://www.okta.com/saml2/service-provider/spkrktcxhbfmkgfelqsv"><BR>
```
Comment (danitornow) [inline saml-idp.py]: Is this a generic okta or specific to Sitka Okta?

**Review by danitornow — APPROVED** [review]
Does this make it only compatible with Okta? If so, might want to call that out in the README.

Everything else looks good without testing it locally.

File: saml-idp.py (hunk 2 above)
Comment (mrmicjam) [reply saml-idp.py]: Specific to an Idp listed in okta settings, this one being for the new "local dev" one I added.  But I see where you are going with this, I should remove the default value so its entered each time :)

File: saml-idp.py (hunk 1 above)
Comment (mrmicjam) [reply saml-idp.py]: good call

File: saml-idp.py (hunk 2 above)
Comment (danitornow) [reply saml-idp.py]: You did read my mind! And if you leave it, just make it clear in the documentation or a comment what it refers to. :D

File: saml-idp.py [hunk 3]
```diff
@@ -149,11 +149,43 @@ def make_assertion(resp, acs, irt, iss, username, validity):
...
+        })
+    uid_val = subelement(uid_attr, "saml2:AttributeValue")
+    uid_val.text=uid
+
     cond = subelement(assertion, 'saml2:Conditions',
                       attrib={'NotBefore': tsnow(-validity),
                               'NotOnOrAfter': tsnow(validity)})
     aud_rest = subelement(cond, 'saml2:AudienceRestriction')
-    subelement(aud_rest, 'saml2:Audience').text = acs  # todo urlparse
+    subelement(aud_rest, 'saml2:Audience').text = sp_entity_id
```
Comment (run-o) [inline saml-idp.py:188]: I don't know what `acs` is but should this replace `acs` or should it be added along with it?

**Review by run-o — APPROVED** [review]
lgtm as far as I can tell. But there's also a lot of stuff I have no clue about in there.

File: saml-idp.py (hunk 3 above)
Comment (mrmicjam) [reply saml-idp.py:188]: I lot of this is a bit mirky for me too... but seems like `acs` here is the url that okta tells the idp to send the client back to with the token, but the Audience is like a unique identifier of the integration on the service provider (okta).  It looks like in some cases they are the same value so thats why the author of this tool used the same variable..  But with okta they are different so i had to make this change..  Which took a few hours to figure out 😬 .  You'll see on line 151 the other place that `acs` is still used.

**Comment by mrmicjam** [comment]
> lgtm as far as I can tell. But there's also a lot of stuff I have no clue about in there.

haha yeah same here

**Pull request merged**